plugins/matrix42/icon_matrix42/connection/connection.py

```diff
@@ -0,0 +1,61 @@
+import insightconnect_plugin_runtime
+from .schema import ConnectionSchema, Input
+import requests
+from insightconnect_plugin_runtime.exceptions import ConnectionTestException
+
+
+class Connection(insightconnect_plugin_runtime.Connection):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+
+    def get_access_token(self):
+        """
+        Get a temporary access token from Matrix42 using the API key.
+        """
+        token_url = self.api_url + "ApiToken/GenerateAccessTokenFromApiToken"
+
+        headers = {"Authorization": f"Bearer {self.api_key}", "Accept": "application/json"}
+
+        try:
+            self.logger.info(f"Requesting access token with API key from {token_url}")
+            response = requests.post(token_url, headers=headers)
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise ConnectionTestException(
+                cause="Failed to retrieve access token from Matrix42.",
+                assistance="Please check your Matrix42 API URL and API key.",
+                data=str(e),
+            )
+
+        return response.json().get("RawToken")
+
+    def connect(self, params):
+        self.logger.info("Connect: Connecting...")
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        self.api_key = params.get(Input.API_KEY).get("secretKey")
+        self.api_url = params.get(Input.API_URL).rstrip("/") + "/"
```

Comment:
Is this rstrip redundant? We should also maybe add .strip() to both inputs.

Reply:
This ensures that the API URL always ends with a single /, regardless of whether the user entered it with or without a trailing slash.

I added a strip() to both lines.